Marlin/src/lcd/menu/menu.cpp

```diff
@@ -347,8 +347,16 @@ void scroll_screen(const uint8_t limit, const bool is_menu) {
 #endif // BABYSTEP_ZPROBE_OFFSET
 
 void _lcd_draw_homing() {
-  constexpr uint8_t line = (LCD_HEIGHT - 1) / 2;
-  if (ui.should_draw()) MenuItem_static::draw(line, GET_TEXT(MSG_LEVEL_BED_HOMING));
+  if (ui.should_draw()) {
+    #if ENABLED(TFTGLCD_PANEL)
+      lcd.clear_buffer();
+      MenuItem_static::draw(4, GET_TEXT(MSG_LEVEL_BED_HOMING));
```

Comment:
Please submit any proposed changes that seem appropriate.

I also find it dubious that `HAS_CHARACTER_LCD` is being enabled for this display. That is only appropriate for HD44780 screens where you send character bytes over SPI to draw strings. I believe that should be disabled and then see what else needs to be changed in response.

Reply:
> I also find it dubious that `HAS_CHARACTER_LCD` is being enabled for this display. That is only appropriate for HD44780 screens where you send character bytes over SPI to draw strings.

This panel was developed precisely as an analogue of a text screen and data transfer to it is performed by text lines. I wrote about this at the beginning. I tried to build the firmware using your changes, but this is not possible as tft libraries are starting to be used that are not involved here. Other dependencies on unused code are also possible. Therefore, for Marlin, this panel is just text. How the information is displayed on the panel does not matter for Marlin. That was the concept. Therefore, `HAS_CHARACTER_LCD` is required for this panel.

Reply:
Well, Marlin has been using `HAS_CHARACTER_LCD` to indicate `ultralcd_HD44780.h` specifically because that is the only character-based display it has historically supported, and the only other option was a DOGM graphical LCD. The meaning has gotten lost with the addition of several displays bringing in other LCD libraries. At this time it seems `HAS_CHARACTER_LCD` is showing its obsolescence.

Since your display provides its own draw interface, it's actually what we would now call a "`MarlinUI`-based display" but not specifically the HD44780 as we commonly know it. So for our purposes we can just treat your new display option as another `MarlinUI` implementation, and later on down the road we'll see what we want to rename `HAS_CHARACTER_LCD` to `HAS_HD44780_LCD`.